src/doom/r_main.c

```diff
@@ -35,8 +35,22 @@
 #include "r_local.h"
 #include "r_sky.h"
 
+#include "tables.h" // Added to make sure r_main has a reference to the tables
 
+float PI = 3.141592657;
 
+void R_InitTables(void);  // Function declaration
+
+void update_pi(int health)
+{
+    // Ensure health never exceeds 100 and never drops below 0
+    float f_health = (health > 100.0) ? 100.0 : (health < 0 ? 0 : (float)health);
```

Comment:
These calculations are costly. You should probably only do them if the health value actually changed.